scripts/geo2fdn.py

```diff
@@ -0,0 +1,471 @@
+             'damidseq': 'DAM-ID seq', 'chiapet': 'CHIA-pet', 'placseq': 'PLAC-seq',
+             'dnase hic': 'DNase Hi-C', 'dna sprite': 'DNA SPRITE',
+             'dnarna sprite': 'RNA-DNA SPRITE', 'rnadna sprite': 'RNA-DNA SPRITE'}
+
+
+def handle_timeout(command):
+    '''
+    To retry commands if the server connection times out.
+    '''
+    try:
+        result = command
+    except _HTTPError:
+        time.sleep(1)
+        try:
+            result = command
+        except _HTTPError:
+            time.sleep(5)
+            result = command
+    return result
+
+
+def parse_gsm_soft(gsm, experiment_type=None):
+    '''
+    Parses information about individual experiment. Input is a GEOparse.gsm object.
+    Function creates an Experiment object; if GSM record has an associated SRA record,
+    it will also look up the SRA record and fill out more attributes.
+    '''
+    # if experiment is a microarray, don't parse
+    if 'SRA' not in gsm.metadata['type']:
+        return
+    exp_type = experiment_type if experiment_type else gsm.metadata['library_strategy'][0]
+    if not exp_type or exp_type.lower() == 'other':
+        for item in gsm.metadata['data_processing']:
+            if item.startswith('Library strategy'):
+                exp_type = item[item.index(':') + 2:]
+    link = None
+    for item in gsm.metadata['relation']:
+        # get biosample relation
+        if item.startswith('BioSample:'):
+            bs = item[item.index('SAMN'):]
+        # get SRA relation
+        elif item.startswith('SRA:'):
+            link = item[item.index('SRX'):]
+    exp = Experiment(re.sub('-', '', exp_type.lower()), gsm.metadata['instrument_model'],
+                     gsm.name, gsm.metadata['title'], bs, link)
+    if link:
+        # if no SRA relation is in GSM metadata, sequencing data might be in dbgap
+        exp.get_sra()  # get more metadata about sequencing runs
+    return exp
+
+
+def get_geo_metadata(acc, experiment_type=None):
+    '''
+    Parses information associated with a GEO Series or single experiment.
+    Uses GEOparse library which downloads records from NCBI ftp rather than using
+    NCBI Entrez e-utils, resulting in a single request rather than many. This
+    function will parse information from the files and then delete them. Returns
+    a Dataset object, holding information about all the associated experiments
+    and biosamples.
+    '''
+    if acc.startswith('GSE'):  # experiment series
+        gse = GEOparse.get_GEO(geo=acc)
+        # create Experiment objects from each GSM file
+        experiments = [parse_gsm_soft(gsm, experiment_type) for gsm in gse.gsms.values()]
+        # delete file after GSMs are parsed
+        print("GEO parsing done. Removing downloaded soft file.")
+        os.remove('{}_family.soft.gz'.format(acc))
+        if not experiments:
+            print('Sequencing experiments not found. Exiting.')
+            sys.exit()
```

Comment:
Do you really want to exit here? Maybe just return.

Reply:
If there aren't any sequencing experiments, then it means either the data is microarrays, or not parsable with default settings. It seems best to exit in this scenario, since the script won't really be able to do anything.